src/sentry/web/frontend/auth_organization_login.py

```diff
@@ -57,6 +58,12 @@ def handle(self, request: Request, organization_slug) -> Response:
 
         request.session.set_test_cookie()
 
+        # check on POST to handle
+        # multiple tabs case well now that we include redirect in url
+        if request.method == "POST":
```

Comment:
Once they log in again, `initiate_login` is called with the path to the page the user was on before they were logged out which takes them back

Reply:
yep! because we look for the `next` URL param in `get_next_uri`.